### DISCUSSION TITLE
Use mutiple db connections for multi tenancy, whats the right way ?

### DISCUSSION THREAD

**Discussion by pullmann4rent** [vitaly-t/pg-promise#906] [Q&A]
Hello,

I am right now developing a multi tenancy SaaS. I use multiple databases and dont know how to use for each connection.
Can I map my connections or should I hard coded and write everytime for each company a new db code in my db.ts file?

```
import pg_promise from 'pg-promise';

export const pgp = pg_promise({

export const { 
  TransactionMode,
  isolationLevel
 } = pgp.txMode;

export const mode = new TransactionMode({
  tiLevel: isolationLevel.serializable,
  readOnly: false,
  deferrable: true
});

export const db1 = pgp({
  host: "XXX",
  user: "XXX",
  database: "XXX",
  password: "xxxx",
  ssl: { rejectUnauthorized: false },
  port: 26232,
  max: 20,
  idleTimeoutMillis: 30000,
  connectionTimeoutMillis: 20000
});

export const db2 = pgp({
  host: "XXX",
  user: "XXX",
  database: "XXX",
  password: "xxxx",
  ssl: { rejectUnauthorized: false },
  port: 26232,
  max: 20,
  idleTimeoutMillis: 30000,
  connectionTimeoutMillis: 20000
});
```

So how can I  start and connect the two db togehter and is there a better way to do this ?

**Comment by vitaly-t** [comment]
See [second parameter `dc` during construction](https://vitaly-t.github.io/pg-promise/), one that can help you differentiate processing within one code.

**Reply by psm-solution-group** [reply]
Thank you very much! Do I get any disadvantages when I use or switch the db connections ? For each tenant I have to switch the connections

**Reply by vitaly-t** [reply]
You would have one `db` object per tenant and use `dc` to share the code. Connections are managed automatically.
